packages/components/src/components/tab-nav/tab-nav.tsx

```diff
@@ -171,10 +185,24 @@ export class TabNav {
   }
 
   selectTab(nextTab: HTMLScaleTabHeaderElement) {
+    this.resetTabs();
+    nextTab.selected = true;
+    this.selectPanel(nextTab);
+  }
+
+  selectPanel(nextTab: HTMLScaleTabHeaderElement) {
+    this.resetPanels();
     const nextPanel = this.findPanelForTab(nextTab);
-    this.reset();
     nextPanel.hidden = false;
-    nextTab.selected = true;
+  }
+
+  deselectTabs(nextTab: HTMLScaleTabHeaderElement) {
```

Comment:
make the current resetTab a general function that can handle this
```suggestion
const tabs = this.getAllEnabledTabs();
    tabs.forEach((tab) => {
      if (!nextTab || tab !== nextTab) {
        tab.selected = false;
      }
    });
```